core/storage/beam_job/gae_models.py

```diff
@@ -24,14 +24,22 @@
 from core import utils
 from core.platform import models
 
+from typing import Dict, Type
+from typing_extensions import Final
+
+MYPY = False
+if MYPY: # pragma: no cover
+    from mypy_imports import base_models
+    from mypy_imports import datastore_services
+
 (base_models,) = models.Registry.import_models([models.NAMES.base_model])
 
 datastore_services = models.Registry.import_datastore_services()
 
-_MAX_ID_GENERATION_ATTEMPTS = 5
+_MAX_ID_GENERATION_ATTEMPTS: Final = 5
 
 
-def _get_new_model_id(model_class: base_models.BaseModel) -> str:
+def _get_new_model_id(model_class: Type[base_models.BaseModel]) -> str:
```

Comment:
Just to check, why did this change?

Reply:
This method is used as a helper function in `get_new_id()`, from there we can find that `model_class` can only accept classes ( not the class's instances ). 
(https://github.com/oppia/oppia/blob/e1469c914cd52971aca5970786b502dd2fd3f520/core/storage/beam_job/gae_models.py#L139 ) 
So that's why `Type[]` is added, but previously `model_class` is annotated with only `base_models.BaseModel` which points to class's instance that is wrong.

Reply:
Ah I see. OK, thanks!